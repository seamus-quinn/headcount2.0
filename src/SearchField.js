import React, { Component } from 'react';
import './styles/SearchField.css';
import PropTypes from 'prop-types';

class SearchField extends Component {
  constructor(props) {
    super(props);
    this.state = {
      inputValue: ''
    };
  }

  handleSearch = (event) => {
    this.setState({inputValue: event.target.value});
    this.props.searchSchools(this.state.inputValue);
  }


  render() {
    return (
      <form className="search-form">
        <input 
          type="text" 
          name="search" 
          placeholder="Search school"
          value={this.state.inputValue}
          onChange={this.handleSearch}
        />
      </form>
    );
  }
}

SearchField.propTypes = {
  findAllMatches: PropTypes.func.isRequired
};

export default SearchField;