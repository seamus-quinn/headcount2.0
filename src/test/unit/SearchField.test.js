import React from 'react';
import ReactDOM from 'react-dom';
import SearchField from '../../SearchField';
import { shallow, mount } from 'enzyme';

describe('SearchField component', () => {
  let wrapper;
  let mockSearchSchools;

  beforeEach(()=>{
    mockSearchSchools = jest.fn();
    wrapper = shallow(<SearchField searchSchools={mockSearchSchools}/>); 
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls searchSchools when user inputs letters', () => {
    wrapper = mount(<SearchField searchSchools={mockSearchSchools}/>);
    const spy = spyOn(wrapper.instance(), 'handleSearch');
    wrapper.instance().forceUpdate();
    const mockEvent = {target: {value: 'as'}};
    wrapper.find('.user-input').simulate('change', mockEvent);

    expect(spy).toHaveBeenCalled();
  });

  it('updates state when handleSearch is called', () => {
    const mockEvent = {target: {value: 'l'}};

    wrapper.instance().handleSearch(mockEvent);

    expect(wrapper.state('inputValue')).toBe(mockEvent.target.value);
  });

  it('should call searchSchools when handleSearch is called', () => {
    const mockEvent = {target: {value: 'l'}};

    wrapper.instance().handleSearch(mockEvent);

    expect(mockSearchSchools).toHaveBeenCalled();
  });
});
