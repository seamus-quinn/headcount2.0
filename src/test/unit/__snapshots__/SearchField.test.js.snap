// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchField
    searchSchools={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="user-input"
        name="search"
        onChange={[Function]}
        placeholder="Search school"
        type="text"
        value=""
/>,
      "className": "search-form",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "user-input",
        "name": "search",
        "onChange": [Function],
        "placeholder": "Search school",
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="user-input"
          name="search"
          onChange={[Function]}
          placeholder="Search school"
          type="text"
          value=""
/>,
        "className": "search-form",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "user-input",
          "name": "search",
          "onChange": [Function],
          "placeholder": "Search school",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
